File: backend/schemas/checkin.py
from datetime import date
from pydantic import BaseModel


class ActivityItem(BaseModel):
    """活动项"""
    title: str
    desc: str


class CheckInCreate(BaseModel):
    """创建签到请求"""
    pass  # 不需要额外字段，user_id 和 checkin_date 从请求中获取


class CheckInResponse(BaseModel):
    """签到响应"""
    id: int
    fortune: str
    good: list[ActivityItem]
    bad: list[ActivityItem]
    streak: int
    checkin_date: str


class CheckInStatus(BaseModel):
    """签到状态"""
    checked_today: bool
    current_streak: int


class CheckInStreak(BaseModel):
    """连续签到统计"""
    longest_streak: int
    current_streak: int

File: frontend/src/types/checkin.ts
export interface ActivityItem {
  title: string;
  desc: string;
}

export interface CheckInResponse {
  id: number;
  fortune: '大吉' | '中吉' | '小吉' | '中平' | '凶' | '大凶';
  good: ActivityItem[];
  bad: ActivityItem[];
  streak: number;
  checkin_date: string;
}

export interface CheckInStatus {
  checked_today: boolean;
  current_streak: number;
}

export interface CheckInStreak {
  longest_streak: number;
  current_streak: number;
}
